### ISSUE TITLE
git.exe using resources even when desktop client is idle

### ISSUE THREAD

**Issue by ebol2000** [desktop/desktop#9675]
### Describe the bug

When I run the desktop client I've been noticing a "run-away" issues with git.exe... see
![image](https://user-images.githubusercontent.com/14001759/80737765-d5022000-8ac8-11ea-937d-fcee247e2804.png)
(note 12 is CPU and the 375 is memory)

This happens only if I launch the desktop client. When I close the client, these aren't cleaned up and I have to kill off these. I'm using Visual Studio and the git add-in there as well, and no git.exe processes are spawned when I use it. I don't use any other Git (other than Bash but that's not running and I rarely use it). I've got multiple repos and so I suspect that's why the multiple git.exe's... but the 12% utilization and memory just keeps going like it's running a fetch or something in a loop.

### Version & OS
Windows 10
Version 2.4.3
Open 'About GitHub Desktop' menu to see the Desktop version. Also include what operating system you are using.

### Steps to reproduce the behavior
Simply launch desktop and let run for a few minutes. Look for CPU utililization... for me git.exe started up and rose to 12%...and stays there even with no issues. Also, relaunching the desktop creates a whole new set of git.exe's and never closes out the original ones. 

### Expected behavior

1. Git.exe's should settle back back to idle over time (not continue to churn in a loop)
2. Git.exe's launched from the desktop should close out (or my expectation) when the desktop closes out.

**Comment by tierninho** [comment]
@ebol2000 Thank you for filing this and bringing it to our attention!

For reference, please see this explanation of the `git.exe` location: https://github.com/desktop/desktop/issues/3708#issuecomment-354665183

By chance are you using [Git for Windows](https://git-for-windows.github.io/)? We recommend installing it and ensuring you check the "add to your `PATH`" option in the installer.

One thing that might be affecting this is when you execute two git commands simultaneously; maybe one from the command prompt and one from an IDE.

Deleting `index.lock` file in your `.git` directory should resolve this.

`rm -f .git/index.lock`

cc @desktop/engineering for thoughts on why so many `git.exe` processes are running.

**Comment by ebol2000** [comment]
@tierninho ... yes, I am using Git for Windows. 
C:\Users\xxx\AppData\Local\GitHubDesktop\app-2.4.3\resources\app\git\mingw64\bin. I had an older version (sorry didn't note the version) of Git for Windows but just installed the latest (2.26.2) and the issue has gone away. I didn't see the Add to Path option (perhaps it was removed) but I did all default (except using Notepad++ as commit editor :) ) options. So I'm happy. Let me know if you want any other info, otherwise let's close this (and I'll be sure to update my Git on a more regular basis). Thanks for the help.
